plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/controller/Assets.js

```diff
@@ -162,6 +168,49 @@ Ext.define('NX.coreui.controller.Assets', {
         assetContainer = me.getAssetContainer();
 
     assetContainer.refreshInfo(gridView.up('grid').componentModel, assetModel);
+  },
+
+  /**
+   * @protected
+   * Enable 'Delete' when user has 'delete' permission.
+   */
+  bindDeleteAssetButton: function(button) {
+    var me = this,
+        component = me.getAssetList().componentModel;
+    button.mon(
+        NX.Conditions.or(
+            NX.Conditions.isPermitted('nexus:repository-admin:*:*:delete'),
+            NX.Conditions.isPermitted('nexus:repository-admin:' + component.get('format') + ':' +
+                component.get('repositoryName') + ':delete')
```

Comment:
I'm not sure you need the OR, unless you found it didn't work otherwise (which could be a bug).

We presently send over the perms in wildcard format asis and have the same logic to determine implied-ness of them in the JS UI as we do on the server.

So if you are granted the *:* version, then asking if you are permitted with the format:repo should do the right thing.